common/tasks/interleaved_weighted_round_robin.go

```diff
@@ -244,7 +262,10 @@ func (s *InterleavedWeightedRoundRobinScheduler[T, K]) eventLoop() {
 		case <-s.notifyChan:
 			s.dispatchTasksWithWeight()
 		case <-checkRateLimiterEnabledTimer.C:
-			s.rateLimiterEnabled.Store(s.options.EnableRateLimiter())
+			s.rateLimiterConfig.Store(rateLimiterConfig{
+				enableThrottle:   s.options.EnableRateLimiter(),
+				enableShadowMode: s.options.EnableRateLimiterShadowMode(),
+			})
```

Comment:
How about we make these fields in the rateLimiterConfig lazy-loaded too? That way we wouldn't need this complexity

Reply:
I.e. change enableThrottle and enableShadowMode to dynamicconfig.BoolPropertyFn

Reply:
Point here is not to evaluate dynamicconfig function in the hot code path of task dispatch.

Reply:
The ideal solution here is to add callback functionality to our dynamic config implementation. It's on our road map, but don't have capacity to work on it currently.

Reply:
fyi, I sped up dynamic config by a lot last year (in https://github.com/temporalio/temporal/pull/3271). It still might not be fast enough for this purpose but might be worth benchmarking. (Callbacks would still be great and useful for other purposes.)